### PR TITLE
fix(client): validate outgoing topic aliases against server limit

### LINKED ISSUE #78: MQTT 5: validate outgoing Topic Aliases against the server's limit
### Problem

In zmqtt 0.2.1, callers can set `PublishProperties.topic_alias`, but the publish path does not check the alias against `Topic Alias Maximum` from CONNACK. Invalid aliases can be sent to brokers that do not accept them.

The relevant paths are [`publish()`](https://github.com/faststream-community/zMQTT/blob/dc9b459244478a36ec9fa14b06ce4607abc297eb/src/zmqtt/client.py#L490-L533) and [`PublishProperties`](https://github.com/faststream-community/zMQTT/blob/dc9b459244478a36ec9fa14b06ce4607abc297eb/src/zmqtt/_internal/packets/properties.py#L247-L275).

### Expected behavior

Implement support for [MQTT 5 §3.2.2.3.8 and §3.3.2.3.4, Topic Alias](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html):

- Reject alias 0 and values outside the valid protocol range before sending PUBLISH.
- Reject aliases greater than the maximum advertised by the server.
- Reject any outgoing alias when Topic Alias Maximum is absent or 0.
- Use the limit negotiated for the current connection and refresh it after reconnect.
- Keep ordinary publications without a Topic Alias working independently of alias support.

### PR DESCRIPTION
**Pull request by AndreyGatsuk**
## Summary

  Implement outgoing Topic Alias validation according to [MQTT 5.0 §3.2.2.3.8 (Topic Alias Maximum) and §3.3.2.3.4 (Topic Alias)](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html).

  Before sending PUBLISH:

  - Reject values outside 1..65535 or above the server's Topic Alias Maximum.
  - Reject aliases when the server omits the maximum or sets it to zero.
  - Use the current connection's limit, including after reconnect.
  - Keep publications without a Topic Alias unaffected.

  Closes #78

  ## Validation

- Added tests for invalid aliases (-1, 0, 65536), missing/zero/exceeded server limits, valid boundary values, updated limits after reconnect, publications without aliases, and MQTT 3.1.1 compatibility (6 test functions, 40 cases).
- Full test suite on Python 3.11 with all five brokers: 732 passed, 24 skipped (existing protocol/broker-specific skips).
- Ruff lint and formatting, mypy, mkdocs build --strict, uv build, and Bandit passed.
- Final docstring and documentation edits passed git diff --check.

- [x] Tests were added or updated when behavior changed
- [x] Documentation was updated when the public API changed
- [x] The PR title follows Conventional Commits, for example `feat(client): add reconnect timeout`

### CODE REVIEW

File: src/zmqtt/client.py [hunk 1]
```diff
@@ -608,6 +613,16 @@ async def publish(
...
+        if properties is not None and properties.topic_alias is not None:
+            alias = properties.topic_alias
+            if not 1 <= alias <= _MAX_TOPIC_ALIAS:
+                msg = "topic_alias must be between 1 and 65535"
+                raise ValueError(msg)
+            connack_properties = self.connection_info.properties
+            maximum = 0 if connack_properties is None else (connack_properties.topic_alias_maximum or 0)
+            if alias > maximum:
+                msg = f"topic_alias {alias} exceeds server Topic Alias Maximum {maximum}"
+                raise ValueError(msg)
```
Comment (borisalekseev) [inline src/zmqtt/client.py:625]: Let's refactor it into `self._validate_publish_properties(properties)

File: tests/test_topic_alias.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        sent = list(transport.sent)
+        with pytest.raises(RuntimeError, match=r"properties require MQTT 5\.0"):
+            await client.publish("test/topic", b"payload", properties=PublishProperties(topic_alias=1))
+        assert transport.sent == sent
+        await client.publish("test/topic", b"without alias")
+
+
+@pytest.mark.parametrize("maximum", [None, 0, 1, 5])
+@pytest.mark.parametrize("resumed", [False, True])
+async def test_reconnect_uses_new_alias_limit(maximum: int | None, resumed: bool) -> None:
```
Comment (borisalekseev) [inline tests/test_topic_alias.py:120]: I think this tests is unnecessary, too complex.

File: tests/test_topic_alias.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        ]
+
+
+async def test_mqtt_v311_still_rejects_publish_properties() -> None:
+    transport = FakeTransport()
+    transport.feed(encode(ConnAck(session_present=False, return_code=0), version="3.1.1"))
+    async with MQTTClient("localhost", transport_factory=transport_factory(transport)) as client:
+        sent = list(transport.sent)
+        with pytest.raises(RuntimeError, match=r"properties require MQTT 5\.0"):
+            await client.publish("test/topic", b"payload", properties=PublishProperties(topic_alias=1))
```
Comment (borisalekseev) [inline tests/test_topic_alias.py:113]: Can we test it using the e2e test case (for less coupling to library internals)?

File: tests/test_topic_alias.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                retain=False,
+                dup=False,
+                properties=properties,
+            )
+        ]
+
+
+@pytest.mark.parametrize("maximum", [None, 0, 3])
+@pytest.mark.parametrize("properties", [None, PublishProperties(content_type="text/plain")])
+async def test_publication_without_alias_is_independent_of_server_limit(
```
Comment (borisalekseev) [inline tests/test_topic_alias.py:86]: What does this test measure? I believe it should be eliminated.

**Review by borisalekseev — CHANGES_REQUESTED** [review]
Too many tests, and the tests are too complicated and coupled with the library's internals. We prefer writing only end-to-end (e2e) tests that measure the public API and behaviour of the library, as these tests are more compatible with refactoring. Unit tests that know about the protocol class and other internal details are allowed only if it is impossible to easily simulate the broker's behaviour in an e2e testing scenario.

I will extend the contribution guide and the project harness soon. Sorry that these concepts weren't explicitly described in the project.
